tools/ZarinPalGateway.php

```diff
@@ -0,0 +1,311 @@
+
+        //if there is an cURL error
+        if ($err) {
+            $this->error = "cURL Error : " . $err;
+
+            return false;
+        } else {
+            /*
+             * Check for verification result
+             */
+            if ($result['Status'] == 100) {
+                return $result['RefID'];
+            } else {
+                $this->error = 'An error occurred while verifying payment : ' . $result['Status'];
+
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Handel ZarinPal errors
+     * Return error message if there is an error
+     *
+     * @return bool|string
+     */
+    public function getPaymentError()
+    {
+        switch ($this->error) {
+            case "-1":
+                return 'اطلاعات ارسال شده ناقص است';
+                break;
+            case "-2":
+                return 'IP و يا مرچنت كد پذيرنده صحيح نيست';
+                break;
+            case "-3":
+                return 'با توجه به محدوديت هاي شاپرك امكان پرداخت با رقم درخواست شده ميسر نمي باشد';
+                break;
+            case "-4":
+                return 'سطح تاييد پذيرنده پايين تر از سطح نقره اي است';
+                break;
+            case "-11":
+                return 'درخواست مورد نظر يافت نشد';
+                break;
+            case "-12":
+                return 'امكان ويرايش درخواست ميسر نمي باشد';
+                break;
+            case "-21":
+                return 'هيچ نوع عمليات مالي براي اين تراكنش يافت نشد';
+                break;
+            case "-22":
+                return 'تراكنش نا موفق مي باشد';
+                break;
+            case "-33":
+                return 'رقم تراكنش با رقم پرداخت شده مطابقت ندارد';
+                break;
+            case "-34":
+                return 'سقف تقسيم تراكنش از لحاظ تعداد يا رقم عبور نموده است';
+                break;
+            case "-40":
+                return 'اجازه دسترسي به متد مربوطه وجود ندارد';
+                break;
+            case "-41":
+                return 'اطلاعات ارسال شده مربوط به AdditionalData غيرمعتبر مي باشد .';
+                break;
+            case "-42":
+                return 'مدت زمان معتبر طول عمر  شناسه  پرداخت بايد بين 30دقيه تا 45روز مي باش';
+                break;
+            case "-54":
+                return 'درخواست مورد نظر آرشيو شده است .';
+                break;
+            case "101":
+                return 'عمليات پرداخت موفق بوده وقبلا PaymentVerification تراكنش انجام شده است';
+                break;
+            case "100":
+                return 'عملیات با موفقیت انجام شد';
+                break;
+        }
+
+        if ( is_string($this->error) ) {
+            return $this->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * Convert IRR to IRT
+     *
+     * @param int|string|float $amount
+     *
+     * @return float|int
+     */
+    public function convertToIrt($amount)
+    {
+        return (int)$amount / 10;
```

Comment:
```suggestion
        return (int) $amount / 10;
```

Reply:
We dont need space between (int) and variable